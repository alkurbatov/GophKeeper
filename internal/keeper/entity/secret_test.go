package entity_test

import (
	"testing"

	"github.com/alkurbatov/goph-keeper/internal/keeper/entity"
	"github.com/gkampitakis/go-snaps/snaps"
)

func TestSecretToString(t *testing.T) {
	tt := []struct {
		name string
		data string
	}{
		{
			name: "Print password",
			data: "1q2w3e",
		},
		{
			name: "Print empty secret",
			data: "",
		},
	}

	for _, tc := range tt {
		t.Run(tc.name, func(t *testing.T) {
			sat := entity.Secret(tc.data)
			snaps.MatchSnapshot(t, sat.String())
		})
	}
}

func TestSecretURIToString(t *testing.T) {
	tt := []struct {
		name     string
		data     string
		expected string
	}{
		{
			name: "Print database URI",
			data: "postgres://postgres:postgres@127.0.0.1:5432/goph?sslmode=disable",
		},
		{
			name: "Print empty secret",
			data: "",
		},
	}

	for _, tc := range tt {
		t.Run(tc.name, func(t *testing.T) {
			sat := entity.SecretURI(tc.data)

			snaps.MatchSnapshot(t, sat.String())
		})
	}
}
