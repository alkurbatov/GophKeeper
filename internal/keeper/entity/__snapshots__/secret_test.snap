
[TestSecretToString/Print_password - 1]
******
---

[TestSecretToString/Print_empty_secret - 1]

---

[TestSecretURIToString/Print_database_URI - 1]
postgres://*****:*****@127.0.0.1:5432/goph?sslmode=disable
---

[TestSecretURIToString/Print_empty_secret - 1]

---
